src/diagnostic/encode.ts

```diff
@@ -15,13 +15,13 @@ export default async function(type: Event): Promise<void> {
             tokens.push(script.data.column);
             tokens.push(script.data.stack);
             tokens.push(script.data.source);
-            queue(tokens);
+            queue(tokens, type);
             break;
         case Event.ImageError:
             if (image.data) {
```

Comment:
When would this if condition not hold?

Reply:
We listen to high level error handler. Browser will fire this event for non image elements too. e.g. canvas. For now I'm restricting to image because I'm not sure what would be a good way to handle other elements. In future we can consider expanding this. At the moment it's focused on image failures.

---
In reply to: [368679267](https://github.com/microsoft/clarity-js/pull/166#discussion_r368679267) [](ancestors = 368679267)

Reply:
Add above as comment. 

---
In reply to: [368807461](https://github.com/microsoft/clarity-js/pull/166#discussion_r368807461) [](ancestors = 368807461,368679267)